sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/ExpressionEvalHelper.scala

```diff
@@ -211,4 +215,80 @@ trait ExpressionEvalHelper {
       plan(inputRow)).get(0, expression.dataType)
     assert(checkResult(actual, expected))
   }
+
+  def checkConsistency(dt: DataType, clazz: Class[_]): Unit = {
```

Comment:
Also, I think that we might be able to clean up the code slightly by adding a type to this method:

```scala
def checkConsistency[E <: Expression: ClassTag](dt: DataType)
```

to let callers write something like

```scala
checkConsistencyBetweenInterpretedAndCodegen[Sinh](DoubleType)
```